NYC Subway Major Incidents (2015 to 2026)

Question: How reliable is the NYC subway, and has that changed over time?

Data: MTA Subway Major Incidents from NY Open Data (data.ny.gov). 
      A "major incident" is an unplanned event that delays 50 or more trains.

Loading the data

In [14]:
import numpy as np

data = np.loadtxt("data.csv", dtype=str, delimiter=",", skiprows=1, encoding="utf-8")
data = np.char.strip(data, '"')

months   = data[:, 0]
lines    = data[:, 2]
daytype  = data[:, 3]
category = data[:, 4]
count    = data[:, 5].astype(int)

years = np.array([m[:4] for m in months])
mon   = np.array([m[5:7] for m in months])

print(data.shape)
print(np.unique(years))

(5791, 6)
['2015' '2016' '2017' '2018' '2019' '2020' '2021' '2022' '2023' '2024'
 '2025' '2026']


Total incidents per year

In [15]:
year_list = np.unique(years)
year_totals = []
for y in year_list:
    year_totals.append(count[years == y].sum())
year_totals = np.array(year_totals)

for i in range(len(year_list)):
    print(year_list[i], year_totals[i])

2015 881
2016 901
2017 904
2018 907
2019 586
2020 322
2021 463
2022 553
2023 534
2024 781
2025 758
2026 517


Which subway line has the most incidents?

In [16]:
line_list = np.unique(lines)
line_list = line_list[line_list != ""]

line_totals = []
for l in line_list:
    line_totals.append(count[lines == l].sum())
line_totals = np.array(line_totals)

order = np.argsort(line_totals)[::-1]
for i in order[:5]:
    print(line_list[i], line_totals[i])

6 676
F 589
2 585
N 583
4 571


Worst and best month of the year

In [17]:
names = ["Jan","Feb","Mar","Apr","May","Jun","Jul","Aug","Sep","Oct","Nov","Dec"]

month_totals = []
for m in np.unique(mon):
    month_totals.append(count[mon == m].sum())
month_totals = np.array(month_totals)

print("Worst month:", names[np.argmax(month_totals)], month_totals.max())
print("Best month:", names[np.argmin(month_totals)], month_totals.min())

Worst month: Jan 810
Best month: Aug 571


What causes the incidents?

In [18]:
cat_list = np.unique(category)
for c in cat_list:
    print(c, count[category == c].sum())

Other 793
Persons on Trackbed/Police/Medical 2091
Signals 2420
Stations and Structure 531
Subway Car 687
Track 1585


Comparing 2019 and 2024 by month

In [19]:
def monthly(year):
    out = []
    for m in ["01","02","03","04","05","06","07","08","09","10","11","12"]:
        out.append(count[(years == year) & (mon == m)].sum())
    return np.array(out)

a = monthly("2019")
b = monthly("2024")

print("2019 (each # = 2 incidents)")
for i in range(12):
    print(names[i], "#" * int(a[i] // 2), a[i])

print()
print("2024 (each # = 2 incidents)")
for i in range(12):
    print(names[i], "#" * int(b[i] // 2), b[i])

print()
print("mean per month 2019:", np.mean(a), " 2024:", np.mean(b))
print("std 2019:", np.std(a), " 2024:", np.std(b))

2019 (each # = 2 incidents)
Jan ############################# 59
Feb ######################### 50
Mar ########################## 53
Apr ########################### 54
May ########################## 52
Jun ####################### 47
Jul ############################### 63
Aug #################### 41
Sep ################## 36
Oct ##################### 42
Nov ################## 37
Dec ########################## 52

2024 (each # = 2 incidents)
Jan ##################################### 74
Feb ############################# 58
Mar ################################## 68
Apr ################################# 66
May ######################### 51
Jun ######################### 51
Jul ################################# 67
Aug ##################################### 74
Sep ################################## 68
Oct ######################### 50
Nov ################################# 66
Dec ############################################ 88

mean per month 2019: 48.833333333333336  2024: 65.08333333333333
std 20

Before vs after 2020

In [20]:
before = year_totals[year_list < "2020"]
after  = year_totals[year_list > "2020"]
print("avg per year 2015-2019:", np.mean(before))
print("avg per year 2021-2025:", np.mean(year_totals[(year_list > "2020") & (year_list < "2026")]))

avg per year 2015-2019: 835.8
avg per year 2021-2025: 617.8


What I found

- Incidents were steady at about 900 a year from 2015 to 2018, then dropped to 586 in 2019 and 322 in 2020 (likely because of COVID), and climbed back to 781 in 2024.
- Comparing 2019 to 2024 by month, the average went from about 49 to 65 incidents per month, which is around 33% more. December 2024 was the worst month in either year with 88.
- The 6 train had the most incidents (676), followed by the F, 2, N, and 4.
- January is the worst month overall and August is the best.
- Signals cause the most incidents (2,420), then people on the trackbed, police, or medical issues (2,091), then track problems (1,585).

Real life connection: (write one sentence about riding the subway here)

2026 only goes through July, so I left it out of the averages.